### DISCUSSION TITLE
Convert Plugin Configuration

### DISCUSSION THREAD

**Discussion by AronAtGit** [beetbox/beets#5346] [Q&A] [answered]
Hey everyone,

Can anybody help me to configure the convert plugin?  What I want is the following:

-Automatically convert to FLAC on import *if* the file is a different lossless format (no lossy to lossless conversions such as MP3 to FLAC.  If the file is lossy I'd like it to import without converting)

-Have two selectable options to convert media that's already in my library: 
1.) convert to AIFF (44.1kHz, 16bit) 
-saved to a new directory like /AIFF/Artist/Album
2.) convert to MP3 (320kbps CBR)
-saved to new directory like /MP3/Artist/Album

Thank you for the assistance!

**Comment by Serene-Arc** [comment, marked as answer]
There are options to do most of what you want but it might require multiple steps. First off, if you never want to convert lossy files, there is an option for that.

Let's go through it in stages:

```yaml
convert:
  never_convert_lossy_files: yes
  auto: yes
  format: flac
```

The above configuration will do the following:
  - Always import lossy files as is
  - Will convert all other files to flac, and then only import the flac, **not** the original file

Now, if you want to convert everything to AIFF, which is in your library, you'd use the `beet convert -f aiff -d /PATH_YOU_WANT/AIFF/ ` with the following in the configuration (you can merge this with the above):

```yaml
convert:
  formats:
    aiff:
      command: ffmpeg -i $source -y -acodec pcm_s16be $dest
      extension: aiff
```

This specifies what AIFF is and what ffmpeg command to use. You can specify the destination path for the conversion, so it goes where you'd like with your library structure being recreated by default. To do MP3 conversion, the command would be `beet convert -f mp3 -d /PATH_YOU_WANT/MP3/`.

However, these command require you to run them, so only the conversion to FLAC happens automatically. Unfortunately, at the current time, the convert plugin doesn't support multiple formats being converted automatically iirc. If you'd like, you can open an enhancement request to enable that.

Hope this helps! As always, run all this on a small test library so that you're absolutely sure it works as intended and you don't ruin your files!
